### PR TITLE
BUILDER-73: Fix issues in generated models

### PR DESCRIPTION
**Pull request by CodrinChirica**
- fix wrong maps, add missing params and remove unnecessary ones

### CODE REVIEW

**Comment by killerboot** [comment]
Are these changes required to generate code out of API definitions?
I'm not sure why strings are worse than objects.

**Comment by CodrinChirica** [comment]
> I'm not sure why strings are worse than objects.

The reason behind it is that it's generating wrong Maps. As you can see here:
https://git.alfresco.com/process-services/alfresco-java-rest-api/blob/develop/generated/alfresco-core-rest-api/src/main/java/com/alfresco/core/model/Activity.java#L41
it generated:
`
private Map<String, String> activitySummary = null;
`
instead of
`
private Map<String, Object> activitySummary = null;
`

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 1]
```diff
@@ -7371,21 +7371,11 @@ definitions:
       properties:
         type: object
         additionalProperties:
-          type: string
+          type: object
       capabilities:
-        $ref: '#/definitions/Capabilities'
-  Capabilities:
```
Comment (gavincornwell) [inline src/main/webapp/definitions/alfresco-core.yaml]: Are we changing this because capabilities are open ended?

Comment (CodrinChirica) [reply src/main/webapp/definitions/alfresco-core.yaml]: Changed to reflect what is here, not to limit it to those 3 values:
https://github.com/Alfresco/alfresco-remote-api/blob/master/src/main/java/org/alfresco/rest/api/model/Person.java#L73

Comment (gavincornwell) [reply src/main/webapp/definitions/alfresco-core.yaml]: Nice, thanks for the clarification.

**Pull request merged**